### PR TITLE
Headless Gamma 

### PR DESCRIPTION
**Pull request by csuvi98**
Added target platforms, Dockerfile, OpenAPI definition and documentation to hu.bme.mit.gamma.api.headless. Arranged files into folders inside hu.bme.mit.gamma.api.headless and hu.bme.mit.gamma.headless.source.generate.

### CODE REVIEW

File: plugins/headless/hu.bme.mit.gamma.headless.source.generate/src/hu/bme/mit/gamma/headless/source/generate/Application.java [hunk 1]
```diff
@@ -36,6 +36,7 @@ public Object start(IApplicationContext context) throws Exception {
 		IProject newProject = workspace.getRoot().getProject(projectName);
 		newProject.create(newProjectDescription, null);
 		newProject.open(null);
+		//ZipFile srcZipFile = new ZipFile(workspace.getRoot().getLocation() + "/" + projectName + ".zip");
```
Comment (grbeni) [inline plugins/headless/hu.bme.mit.gamma.headless.source.generate/src/hu/bme/mit/gamma/headless/source/generate/Application.java:39]: What is the purpose of this?

File: plugins/xsts/hu.bme.mit.gamma.statechart.lowlevel.model/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -1,6 +1,6 @@
 Manifest-Version: 1.0
 Bundle-ManifestVersion: 2
-Bundle-Name: Gamma Low-level Statechart Language Metamodel
```
Comment (grbeni) [inline plugins/xsts/hu.bme.mit.gamma.statechart.lowlevel.model/META-INF/MANIFEST.MF:3]: Let us keep the original name.

File: plugins/headless/hu.bme.mit.gamma.api.headless/docs/headless-gamma-document.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+ 8. Open a terminal and build the docker image with the following command: `docker build -t gamma <path to working folder>` (the working folder is the folder created in the first step) . This will create a docker image named "gamma".
+ 9. In a terminal, run a docker container from the gamma image using the `docker run -it -p 8080:8080 --network host --name gamma_container gamma:latest` command. This command will bind the localhost:8080 address of the host machine to the Docker container, forwarding commands to the webserver running inside. The container is named "gamma_container", and starts in interactive mode, allowing for CLI access. The last parameter is the image, which is `gamma:latest`, which indicates that the latest Gamma image build is used.
+ 
+It is possible to change the port binding if the user wishes, but the webserver listens to port 8080. In the port binding parameter, the part before the ":" stands for the host port, and the one after is the container port. So changing the parameter to "5555:8080" is valid, but "8080:5555" would result in communication failure.  Names of the image and container can also be changed.
+
+
+
+<!--stackedit_data:
```
Comment (grbeni) [inline plugins/headless/hu.bme.mit.gamma.api.headless/docs/headless-gamma-document.md:145]: What is this?

File: plugins/headless/hu.bme.mit.gamma.api.headless/api/gamma-wrapper.yaml [hunk 4]
```diff
@@ -0,0 +1,236 @@
+openapi: 3.0.0
+info:
+  version: 1.0.0
+  title: Gamma Wrapper API
+  description: 'An API that allows users to use the Gamma framework via http requests'
+  contact:
+    name: Csortán Szilárd
```
Comment (grbeni) [inline plugins/headless/hu.bme.mit.gamma.api.headless/api/gamma-wrapper.yaml:7]: These should be changed.

File: plugins/headless/hu.bme.mit.gamma.headless.source.generate/src/hu/bme/mit/gamma/headless/source/generate/Application.java (hunk 1 above)
Comment (csuvi98) [reply plugins/headless/hu.bme.mit.gamma.headless.source.generate/src/hu/bme/mit/gamma/headless/source/generate/Application.java:39]: Probably a mistake, will be removed.

**Pull request merged**